packages/flutter_tools/lib/src/protocol_discovery.dart

```diff
@@ -60,10 +59,14 @@ class ProtocolDiscovery {
 
   void _handleLine(String line) {
     Uri uri;
-    final int index = line.indexOf(_prefix + 'http://');
-    if (index >= 0) {
+
+    final RegExp r =
+        new RegExp('${RegExp.escape(serviceName)} (.*) (http://[^ \n]+)');
```

Comment:
Why `(.*)` instead of `listening on `?

Reply:
I preferred to be more wide in the search, so whatever jumps there (even if in the future something changes), btw we can force to listening on without particular issue